errai-ui/src/main/java/org/jboss/errai/ui/rebind/TranslationServiceGenerator.java

```diff
@@ -289,7 +289,8 @@ private String getPathRoot(final MetaClass bundleClass, final URL resource) {
       // Do NOT use File.separatorChar here: Url.getPath() always uses forward-slashes
       relativePath = bundleClass.getPackageName().replace('.', '/');
 
-    return fullPath.substring(0, fullPath.indexOf(relativePath));
+    String pathRoot = fullPath.substring(0, fullPath.indexOf(relativePath));

```

Comment:
I attempt to preserve the ```jar:file:/path/to/jar.jar!/``` semantics.. I we remove the JAR ```URL``` "wrapping" we'd also need to strip the trailing ```!``` e.g. ```jar:file:/path/to/jar.jar!/``` would become ```file:/path/to/jar.jar```.